metadata/metadata.go

```diff
@@ -111,14 +111,27 @@ func Join(mds ...MD) MD {
 type mdIncomingKey struct{}
 type mdOutgoingKey struct{}
 
-// NewIncomingContext creates a new context with incoming md attached.
+// NewIncomingContext creates a new context with incoming md attached. If used
+// in conjunction with AppendToOutgoingContext, NewIncomingContext will
```

Comment:
This is not right -- incoming metadata and outgoing metadata are separate.  This comment should be on `NewOutgoingContext`.

Reply:
Ah, shoot, more leftovers from before. Thanks for spotting

Reply:
Done. Also changed the verbiage of the "intended for grpc use only" note above `FromOutgoingContextRaw` to match other notes in the package.

Reply:
I liked the comment, it was just in the wrong place.  Can you put a similar note on the `NewOutgoingContext` doc string?  I.e. "Any existing outgoing metadata in the context is overwritten".

Reply:
Wow, serious monday brain not reading the second part of that. Done - moved to NewOutgoingContext